### PR TITLE
Add kwargs

### PR DESCRIPTION
**Pull request by RenanOD**
Solves issue #17.

### CODE REVIEW

**Comment by RenanOD** [comment]
Solvers `JSOSolvers.tron` and `JSOSolvers.lbfgs` don't take kwargs so I didn't test `fminbnd` and `fminunc`. I guess I could just specify :ipopt though...

File: src/fminunc.jl [hunk 1]
```diff
@@ -14,9 +14,9 @@ function fminunc(f, x :: AbstractVector;
                  max_eval :: Int = -1,
                  max_time :: Float64 = 30.0,
                  verbose :: Bool = false,
-                 logger = verbose ? ConsoleLogger() : NullLogger()
+                 logger = verbose ? ConsoleLogger() : NullLogger(), kwargs...
```
Comment (abelsiqueira) [inline src/fminunc.jl]: New line

File: test/fminbnd.jl [hunk 2]
```diff
@@ -8,6 +8,12 @@ function fminbnd_tests()
     @test norm(output.solution .- [0.5; 0.25]) < 1e-6
     @test output.dual_feas < 1e-6
 
+    kwargs = (x0 = [-3.0; 1.0],)
+
+    output = fminbnd(f, x₀, ℓ, u; solver = :ipopt, kwargs...)
```
Comment (abelsiqueira) [inline test/fminbnd.jl]: Don't create a `kwargs`, manually pass the argument.
Notice that `x0` is not a valid argument for IPOPT, try
```julia
output = fminbnd(f, x₀, ℓ, u; solver = :ipopt, nlp_scaling_method="none")
```

File: test/linprog.jl [hunk 3]
```diff
@@ -43,9 +43,10 @@ function linprog_tests()
     @test output.primal_feas < 1e-6
     @test norm(output.solution - sol) < 1e-6
 
-    x0 = [1.; -2.]
+    kwargs = (x0 = [1.; -2.],)
 
-    output = linprog(c, Aineq = opAineq, blow = blow, bupp = bupp, Aeq = opAeq, beq = beq, lvar = lvar, uvar = uvar, x0 = x0)
+    output = linprog(c, Aineq = opAineq, blow = blow, bupp = bupp, Aeq = opAeq, beq = beq,
+                     lvar = lvar, uvar = uvar; kwargs...)
```
Comment (abelsiqueira) [inline test/linprog.jl]: The objective is to pass extra keyword arguments to the solver, so keep `x0` and use a different keyword argument. See the other exameple.

File: test/runtests.jl [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # this package
-using JSOSuite
+using Main.JSOSuite
```
Comment (abelsiqueira) [inline test/runtests.jl]: Why is this needed?

**Comment by abelsiqueira** [comment]
`lbfgs`, `trunk` and `tron` all have additional keyword arguments. See them [here](https://github.com/JuliaSmoothOptimizers/JSOSolvers.jl/tree/master/src).

File: test/runtests.jl (hunk 4 above)
Comment (RenanOD) [reply test/runtests.jl]: Bad habit :confused:

File: test/fminbnd.jl (hunk 2 above)
Comment (RenanOD) [reply test/fminbnd.jl]: It isn't? I looked in [NLPModelsIpopt](https://github.com/JuliaSmoothOptimizers/NLPModelsIpopt.jl/blob/master/src/NLPModelsIpopt.jl)

Comment (RenanOD) [reply test/fminbnd.jl]: Oh I get it now.

File: test/fminbnd.jl [hunk 5]
```diff
@@ -8,9 +8,7 @@ function fminbnd_tests()
     @test norm(output.solution .- [0.5; 0.25]) < 1e-6
     @test output.dual_feas < 1e-6
 
-    kwargs = (x0 = [-3.0; 1.0],)
-
-    output = fminbnd(f, x₀, ℓ, u; solver = :ipopt, kwargs...)
+    output = fminbnd(f, x₀, ℓ, u; solver = :tron, max_eval = Int(1e6))
```
Comment (abelsiqueira) [inline test/fminbnd.jl]: `max_eval` is an argument already accepted and treated by `fminbnd`. It has to be an argument for `tron` that is not an argument for `fminbnd`.

File: test/fminunc.jl [hunk 6]
```diff
@@ -7,9 +7,7 @@ function fminunc_tests()
     @test norm(output.solution .- 1) < 1e-6
     @test output.dual_feas < 1e-6
 
-    kwargs = (x0 = [-3.0; 1.0],)
-
-    output = fminunc(f, x₀; solver = :ipopt, kwargs...)
+    output = fminunc(f, x₀; solver = :lbfgs, max_eval = Int(1e6))
```
Comment (abelsiqueira) [inline test/fminunc.jl]: See other

**Comment by abelsiqueira** [comment]
Thanks
